contentcuration/contentcuration/static/js/edit_channel/new_channel/views.js

```diff
@@ -163,10 +135,22 @@ var ChannelListPage  = BaseViews.BaseView.extend({
 	},
 	set_active_channel(channel) {
 		this.$el.removeClass("active_channel");
-		this.starred_channel_list.set_active_channel(channel);
-		this.current_channel_list.set_active_channel(channel);
-		this.public_channel_list.set_active_channel(channel);
-		this.viewonly_channel_list.set_active_channel(channel);
+		// This function gets called by open_channel, but since the API calls to populate the channels list can be slow,
+		// these lists may be undefined when this gets called. For now, just check that the list is undefined to prevent
+		// this from throwing errors.
+		// TODO: The real fix is to make the channel list code more performant so that this doesn't happen.
```

Comment:
Shouldn't we only be calling `set_active_channel` when all the promises resolve?

Reply:
The problem is that open_channel, and hence set_active_channel, is being triggered by user interactions, and the page is interactive before all the promises have resolved. e.g. you might see your public channels, and click open_channel on one of them, but your bookmarked channels haven't loaded yet, giving you an error when you try to set the active channel in the bookmarked channels view.

I think the real fix will need some discussion, as there are a couple things I'm not so sure about going on here. One is that we're setting the state on four different tab views at once (even ones the user has never loaded yet), when those tabs probably should be reading the current state on load instead. The second is the question of whether each channels tab should be its own separate view. 

In the meantime, though, the set_active_channel errors are one of the most frequently appearing Sentry errors (well over 100 reports and climbing), so I felt this would be a simple fix that would significantly improve the Sentry signal-to-noise ratio.

Reply:
Ah, sorry, bookmarked = starred